legend-engine-xt-serviceStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/service/ServiceExecutor.java

```diff
@@ -77,41 +85,28 @@ public static InputStreamResult executeHttpService(String url, List<Header> head
             throw new RuntimeException(errMsg, e);
         }
 
-        InputStream response = executeRequest(httpMethod, uri, headers, requestBodyEntity, mimeType, securitySchemes, profiles);
+        InputStream response = executeRequest(httpMethod, uri, headers, requestBodyEntity, mimeType, securitySchemes, authenticationSchemeRequirements, profiles);
         return new InputStreamResult(response, org.eclipse.collections.api.factory.Lists.mutable.with(new ServiceStoreExecutionActivity(url)));
     }
 
-    public static InputStream executeRequest(HttpMethod httpMethod, URI uri, List<Header> headers, StringEntity requestBodyDescription, String mimeType, List<SecurityScheme> securitySchemes, MutableList<CommonProfile> profiles)
+    public InputStream executeRequest(HttpMethod httpMethod, URI uri, List<Header> headers, StringEntity requestBodyDescription, String mimeType, List<SecurityScheme> securitySchemes, List<AuthenticationSchemeRequirement> authenticationSchemeRequirements, MutableList<CommonProfile> profiles)
     {
         Span span = GlobalTracer.get().activeSpan();
 
-        HttpUriRequest request;
-        switch (httpMethod)
-        {
-            case GET:
-                request = new HttpGet(uri);
-                break;
-            case POST:
-                HttpPost postRequest = new HttpPost(uri);
-                postRequest.setEntity(requestBodyDescription);
-                request = postRequest;
-                break;
-            default:
-                throw new UnsupportedOperationException("The HTTP method " + httpMethod + " is not supported");
-        }
-        ListIterate.forEach(headers, header -> request.addHeader(header));
-
         try
         {
-            HttpClientBuilder clientBuilder = HttpClients.custom();
-
-            if (securitySchemes != null)
+            ServiceStoreConnectionProvider serviceStoreConnectionProvider = new ServiceStoreConnectionProvider(this.credentialProviderProvider);
+            ServiceStoreConnectionSpecification connectionSpecification = new ServiceStoreConnectionSpecification(uri, httpMethod.toString(), headers, requestBodyDescription, mimeType);
+            if (authenticationSchemeRequirements == null)
             {
-                securitySchemes.forEach(securityScheme -> processSecurityScheme(clientBuilder, profiles, securityScheme));
+                authenticationSchemeRequirements = Lists.mutable.empty();
             }
+            ServiceStoreAuthenticationSpecification authenticationSpecification = new ServiceStoreAuthenticationSpecification(authenticationSchemeRequirements,securitySchemes);
```

Comment:
nitpick : Can we move this null inside `ServiceStoreAuthenticationSpecification`?

Reply:
done